sklearn/decomposition/kernel_pca.py

```diff
@@ -300,8 +324,16 @@ def transform(self, X):
         """
         check_is_fitted(self, 'X_fit_')
 
+        # Compute centered gram matrix between X and training data X_fit_
         K = self._centerer.transform(self._get_kernel(X, self.X_fit_))
-        return np.dot(K, self.alphas_ / np.sqrt(self.lambdas_))
+
+        # scale eigenvectors (properly account for null-space for dot product)
+        nz = np.flatnonzero(self.lambdas_)
```

Comment:
nitpick: can you use `non_zero` instead of `nz` (only if it doesn't make it worse with the long lines)

Reply:
What about `non_zeros` with an `s`? To mean the non zero indices

Reply:
yes that's better

Reply:
thanks!

Reply:
Note : that makes the line grow above 80 chars so I used a parenthesis construct to go multiline, rather than a backslash ; if I remember well that's what sklearn tends to prefer, do you confirm ?

```python
scaled_alphas[:, non_zeros] = (self.alphas_[:, non_zeros]
                               / np.sqrt(self.lambdas_[non_zeros]))
```

Reply:
yes parenthesis are preferred

Reply:
perfect, thanks!